tracer/test/Datadog.Trace.Tools.Runner.IntegrationTests/BaseRunCommandTests.cs

```diff
@@ -0,0 +1,122 @@
+
+            Program.CallbackForTests = (c, a, e) =>
+            {
+                command = c;
+                arguments = a;
+                environmentVariables = e;
+                callbackInvoked = true;
+            };
+
+            // CI visibility mode checks if there's a running agent
+            using var agent = EnableCiVisibilityMode ? new MockTracerAgent(TcpPortProvider.GetOpenPort()) : null;
+
+            var agentUrl = $"http://localhost:{agent?.Port ?? 1111}";
+
+            var commandLine = $"{CommandPrefix} test.exe --dd-env TestEnv --dd-service TestService --dd-version TestVersion --tracer-home TestTracerHome --agent-url {agentUrl} --set-env VAR1=A --set-env VAR2=B";
+
+            using var console = ConsoleHelper.Redirect();
+
+            var exitCode = Program.Main(commandLine.Split(' '));
+
+            using var scope = new AssertionScope();
+
+            scope.AddReportable("output", console.Output);
+
+            exitCode.Should().Be(0);
+            callbackInvoked.Should().BeTrue();
+
+            command.Should().Be("test.exe");
+            arguments.Should().BeNullOrEmpty();
+            environmentVariables.Should().NotBeNull();
+
+            environmentVariables["DD_ENV"].Should().Be("TestEnv");
+            environmentVariables["DD_SERVICE"].Should().Be("TestService");
+            environmentVariables["DD_VERSION"].Should().Be("TestVersion");
+            environmentVariables["DD_DOTNET_TRACER_HOME"].Should().Be("TestTracerHome");
+            environmentVariables["DD_TRACE_AGENT_URL"].Should().Be(agentUrl);
+            environmentVariables["VAR1"].Should().Be("A");
+            environmentVariables["VAR2"].Should().Be("B");
+
+            if (EnableCiVisibilityMode)
+            {
+                environmentVariables["DD_CIVISIBILITY_ENABLED"].Should().Be("1");
+            }
+            else
+            {
+                environmentVariables.Should().NotContainKey("DD_CIVISIBILITY_ENABLED");
+            }
+        }
+
+        [Fact]
+        public void AdditionalArguments()
+        {
+            string command = null;
+            string arguments = null;
+            Dictionary<string, string> environmentVariables = null;
+            bool callbackInvoked = false;
+
+            Program.CallbackForTests = (c, a, e) =>
+            {
+                command = c;
+                arguments = a;
+                environmentVariables = e;
+                callbackInvoked = true;
+            };
+
+            // CI visibility mode checks if there's a running agent
+            using var agent = EnableCiVisibilityMode ? new MockTracerAgent(TcpPortProvider.GetOpenPort()) : null;
+
+            var agentUrl = $"http://localhost:{agent?.Port ?? 1111}";
+
+            // dd-env is an argument for the target application and therefore shouldn't set the DD_ENV variable
+            var commandLine = $"{CommandPrefix} --tracer-home dummyFolder --agent-url {agentUrl} -- test.exe --dd-env test";
```

Comment:
Took me a while to figure this one. I wonder should (can) we require that the command arguments e.g. `test.exe` are always placed last? Similar to how docker does it? So the command in the previous test should be written as 

```csharp
            var commandLine = $"{CommandPrefix} --dd-env TestEnv --dd-service TestService --dd-version TestVersion --tracer-home TestTracerHome --agent-url {agentUrl} --set-env VAR1=A --set-env VAR2=B test.exe";
```

(or, equivalent)


```csharp
            var commandLine = $"{CommandPrefix} --dd-env TestEnv --dd-service TestService --dd-version TestVersion --tracer-home TestTracerHome --agent-url {agentUrl} --set-env VAR1=A --set-env VAR2=B -- test.exe";
```

Reply:
Also I need to add a few examples to the help, especially for the `dd-trace run -- <command>` that is hard to discover when you're not used to it